projects/packages/analyzer/scripts/jp-analyze-parallel.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+ini_set( 'memory_limit', '512M' );
+
+require dirname( __DIR__ ) . '/vendor/autoload.php';
+
+use Automattic\Jetpack\Analyzer\Locker;
+use Automattic\Jetpack\Analyzer\Model;
+
+$slurper_path = dirname( __DIR__ ) . '/slurper/plugins';
+
+$jetpack_exclude = array( '.git', 'vendor', 'tests', 'docker', 'bin', 'scss', 'images', 'docs', 'languages', 'node_modules' );
+// $jetpack_new_path = '/Users/brbrr/Developer/a8c/jetpack/projects/plugins/jetpack';
+$jetpack_new_path = '/var/www/html/wp-content/plugins/jetpack';
+$jetpack_old_path = $slurper_path . '/jetpack-production';
+
+class ScanManager {
+	private $count          = 0;
+	private $procs          = array();
+	private $pipez          = array();
+	private $descriptorspec = array(
+		0 => array( 'pipe', 'r' ),
+		1 => array( 'pipe', 'w' ),
+		2 => array( 'file', '/tmp/error-output.txt', 'a' ),
+	);
+
+	private $model;
+
+	public function __construct() {
+		$this->model = new Model();
+	}
+
+	public function start_proc( $folder_name ) {
+			$this->wait_for_procs();
+			echo 'Starting ' . basename( $folder_name ) . "\n";
+			$cmd = 'php ' . dirname( __DIR__ ) . '/scripts/jp-warnings-job.php ' . $folder_name;
+			$this->count++;
+			$process       = proc_open( $cmd, $this->descriptorspec, $pipes );
+			$this->procs[] = $process;
+			$this->pipez[] = $pipes;
+		$this->model->update_process( 0, basename( $folder_name ) );
+
+	}
+
+	public function check_procs() {
+		foreach ( $this->procs as $id => $proc ) {
+			$status = proc_get_status( $proc );
+			if ( $status['pid'] > 0 ) {
```

Comment:
This doesn't seem like it'll do what you intend. It'll go through the list and for each process that has _started_ running it will wait until it _exits_. If each process produces more than a small amount of output (a few thousand bytes, probably), it'll block until its turn comes which seems like it would defeat the parallelism.

It seems more likely you're either wanting to process only those that have _finished_ running (`! $status['running']`?), or that you want to read data from all open processes with a select loop of some sort until at least one finishes.